### DISCUSSION TITLE
"Render" math formulas in markdown files

### DISCUSSION THREAD

**Discussion by firubat** [LazyVim/LazyVim#4551] [Q&A]
I'd like to have a similar experience to vimtex's "rendering" of math, but in markdown files.
The `render-markdown.nvim` plugin gives me horrible results as the "rendered" formula appears "above" the actual code, but with a few lines of space.

I'd like to simply use vimtex's conceal rules, but I couldn't figure out how to do that. 
I tried to follow [this discussion](https://github.com/lervag/vimtex/issues/2874), adding the code @lervag [suggested](https://github.com/lervag/vimtex/issues/2874#issuecomment-2226373920) to `/syntax/markdown.tex` or `after/syntax/markdown.tex` but it had no effect.

Any help appreciated.

**Comment by lervag** [comment]
Notice that the solution you refer to is missing an important point: VimTeX's syntax highlighting relies on the legacy syntax system, and if you want to use it for math in Markdown, then you also need to use the legacy syntax enging for Markdown. Essentially, you need to configure nvim-treesitter to ignore markdown similar to how you ignore latex, e.g.:

```lua
require("nvim-treesitter.configs").setup {
  -- …
  highlight = {
    enable = true,
    disable = { "latex", "markdown" },
  },
  -- …
}
```
